server/server_test.go

```diff
@@ -274,6 +276,121 @@ func TestSubmitCertificateVerification(t *testing.T) {
 	require.NoError(t, eg.Wait())
 }
 
+func createTestKeyFile(t *testing.T, dir, name string, data []byte) string {
+	path := filepath.Join(dir, name)
+	if err := os.WriteFile(path, data, 0o644); err != nil {
+		t.Fatalf("failed to write key file: %v", err)
+	}
```

Comment:
Also this function can be inlined 🙂 

```suggestion
	require.NoError(t, os.WriteFile(path, data, 0o644))
```